src/quic.c

```diff
@@ -774,6 +780,11 @@ int wolfSSL_quic_receive(WOLFSSL* ssl, byte* buf, word32 sz)
         n = 0;
         if (ssl->quic.input_head) {
             n = quic_record_transfer(ssl->quic.input_head, buf, sz);
+
+            /* record too small to be fit into a RecordLayerHeader struct. */
+            if (n == -1) {
```

Comment:
`n` is unsigned here -- but there's a bigger problem, `quic_record_transfer()` is returning unsigned (`word32`), but it's returning `-1` to signify error.

so you'll need to fix `quic_record_transfer()` to return `sword32`, fix signedness clashes that arise from that, and change `n` in `wolfSSL_quic_receive()` to be an `sword32` and fix clashes that arise from _that_.

I'm quite surprised none of the analyzers are warning us about `return -1` from `word32 quic_record_transfer()`.